WeakAuras/BuffTrigger2.lua

```diff
@@ -2518,6 +2517,11 @@ function BuffTrigger.Add(data)
 
       local combineMode = "showOne"
       local perUnitMode
+
+      if WeakAuras.IsUntrackableSoftTarget(trigger.unit) then
```

Comment:
This needs a bit of tweaks, because:

* The second parameter to UpdateWarnings is supposed to be a "id", that is unique per warning.
* You want to clear the warning if no trigger uses a soft target, so you'd want to to iterate over all triggers. And if any of them usees a untrackable soft target unit, you run your current code, else you run: Private.AuraWarnings.UpdateWarning(data.uid, "warningId"), that clears the warnings for that id.